src/Reducer/SavedReducer.js

```diff
@@ -0,0 +1,12 @@
+export function SavedReducer(state,action){
+    switch(action.type){
+        case "ADD-TO-SAVED":
+            return {...state,savedvideos:[...state.savedvideos,{...action.payload,isSaved:true}]}
+        case "REMOVE-FROM-SAVED":
+            return {...state,savedvideos:state.savedvideos.filter((video)=>video.id !== action.payload.id)}
+        case "REFRESH-STATE":
+            return action.payload
```

Comment:
You should always send an object in a payload, it makes it easy to understand what data in coming